### PR TITLE
replace time.h with ctime and switch to std:: variants

### PR DESCRIPTION
**Pull request by neheb**
Made similar std adjustments in edited files.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/LogBackend.cxx [hunk 1]
```diff
@@ -90,10 +89,10 @@ EnableLogTimestamp() noexcept
 static const char *
 log_date() noexcept
 {
-	static constexpr size_t LOG_DATE_BUF_SIZE = 16;
+	static constexpr std::size_t LOG_DATE_BUF_SIZE = 16;
 	static char buf[LOG_DATE_BUF_SIZE];
-	time_t t = time(nullptr);
-	strftime(buf, LOG_DATE_BUF_SIZE, "%b %d %H:%M : ", localtime(&t));
+	std::time_t t = std::time(nullptr);
+	std::strftime(buf, LOG_DATE_BUF_SIZE, "%b %d %H:%M : ", localtime(&t));
```
Comment (MaxKellermann) [inline src/LogBackend.cxx:95]: This should be `std::localtime`

File: src/time/Convert.cxx [hunk 2]
```diff
@@ -57,7 +54,7 @@ LocalTime(std::chrono::system_clock::time_point tp)
 {
 	const time_t t = std::chrono::system_clock::to_time_t(tp);
 #ifdef _WIN32
-	const struct tm *tm = localtime(&t);
+	const struct tm *tm = std::localtime(&t);
 #else
 	struct tm buffer, *tm = localtime_r(&t, &buffer);
```
Comment (MaxKellermann) [inline src/time/Convert.cxx:59]: Which header provides `localtime_r` now?

File: src/time/Convert.cxx [hunk 3]
```diff
@@ -80,7 +77,7 @@ GetTimeZoneOffset() noexcept
 	struct tm tm;
```
Comment (MaxKellermann) [inline src/time/Convert.cxx:77]: Which header provides `struct tm` now?

Comment (neheb) [reply src/time/Convert.cxx:77]: There's std::tm. I'll convert.

File: src/time/Convert.cxx (hunk 2 above)
Comment (neheb) [reply src/time/Convert.cxx:59]: I believe ctime includes time.h. I'll try to find a reference from the standard.

File: src/LogBackend.cxx (hunk 1 above)
Comment (neheb) [reply src/LogBackend.cxx:95]: I'll fix it.

**Comment by neheb** [comment]
On second thought, never mind. After doing some research, it seems there are no good alternatives to localtime and gmtime untill C++20. I'll defer it to when this project moves to that.

**Pull request closed without merging**